src/kudu/consensus/leader_election.cc

```diff
@@ -567,12 +977,28 @@ void LeaderElection::VoteResponseRpcCallback(const std::string& voter_uuid) {
   CheckForDecision();
 }
 
-void LeaderElection::RecordVoteUnlocked(const VoterState& state, ElectionVote vote) {
+void LeaderElection::RecordVoteUnlocked(
+    const VoterState& state, ElectionVote vote) {
   DCHECK(lock_.is_locked());
 
+  // Construct vote information struct.
+  VoteInfo vote_info;
+  vote_info.vote = vote;
+  for (int i = 0; i < state.response.previous_vote_history_size(); i++) {
+    vote_info.previous_vote_history.push_back(
+        state.response.previous_vote_history(i));
+  }
+
+  // Sorting according to election_term.
+  std::sort(
+      vote_info.previous_vote_history.begin(),
+      vote_info.previous_vote_history.end(),
+      compare_PreviousVotePB);
+
   // Record the vote.
   bool duplicate;
-  Status s = vote_counter_->RegisterVote(state.peer_uuid, vote, &duplicate);
+  Status s = vote_counter_->RegisterVote(
+      state.peer_uuid, vote_info, &duplicate);
```

Comment:
Either  std::move vote_info or make registerVote work on a const reference.

```suggestion
      state.peer_uuid, std::move(vote_info), &duplicate);
```